ansible_base/authentication/authenticator_plugins/saml.py

```diff
@@ -289,8 +289,16 @@ def extra_data(self, user, backend, response, *args, **kwargs):
             if perm in attrs:
                 kwargs["social"].extra_data[perm] = attrs[perm]
 
-        # Move group spec up a level if present
-        if "Group" in attrs:
+        # Get configured group attribute, if present
+        configuration = self.database_instance.configuration if hasattr(self.database_instance, 'configuration') else None
+        idp_groups_attribute_name = self.configuration_class.settings_to_enabled_idps_fields['IDP_GROUPS']
+        configured_groups_attribute = configuration['ENABLED_IDPS'][idp_string].get(idp_groups_attribute_name) if configuration is not None else None
```

Comment:
You could simplify some of this logic here:
```suggestion
        configuration = getattr(self.database_instance, 'configuration', {})
        idp_groups_attribute_name = self.configuration_class.settings_to_enabled_idps_fields.get('IDP_GROUPS', None)
        configured_groups_attribute = configuration.get('ENABLED_IDPS', {}).get(idp_string, {}).get(idp_groups_attribute_name, None)
```